optuna/progress_bar.py

```diff
@@ -75,15 +77,21 @@ def _init_valid(self) -> None:
         optuna_logging.disable_default_handler()
         optuna_logging._get_library_root_logger().addHandler(_tqdm_handler)
 
-    def update(self, elapsed_seconds: float) -> None:
+    def update(self, elapsed_seconds: float, study: "Study") -> None:
         """Update the progress bars if ``is_valid`` is :obj:`True`.
 
         Args:
             elapsed_seconds:
                 The time past since :func:`~optuna.study.Study.optimize` started.
+            study:
+                The current study object.
         """
 
         if self._is_valid:
+            if not study._is_multi_objective() and study.get_trials(states=[TrialState.COMPLETE]):
```

Comment:
This line seems to be slowing down the process.

Reply:
This PR seems to be several times slower than master.

Reply:
Thank you for your comments! Indeed, the original code was severely slower than the master branch. I've updated this part.